### PR TITLE
Add bean factory example and docs

### PR DESCRIPTION
**Pull request by smyrick**
### :pencil: Description
Add new docs explaining how the use Spring `BeanFactory` for nested queries

Screen Shot 2020-04-10 at 3 58 23 PM

### :link: Related Issues
https://github.com/ExpediaGroup/graphql-kotlin/issues/655

### CODE REVIEW

**Comment by smyrick** [comment]
New format

Screen Shot 2020-04-10 at 4 40 28 PM

File: docs/spring-server/spring-bean-factory.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+---
+id: spring-bean-factory
+title: Spring BeanFactory
+---
+
+You may want to have schema inputs at different levels and if you are using Spring components that expect these inputs to be Autowired you will have to use [BeanFactory](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/factory/BeanFactory.html).
+
+There is an example of how to set this up in [TopLevelBeanFactoryQuery.kt](https://github.com/ExpediaGroup/graphql-kotlin/blob/master/examples/spring/src/main/kotlin/com/expediagroup/graphql/examples/query/TopLevelBeanFactoryQuery.kt) however you can also use the alternate approach of adding the [DataFetchingEnvironment](../writing-schemas/nested-queries.md) to access the arguments in parent nodes.
+
+Both methods have their pros and cons but you may find that the `DataFetchingEnvironment` approach means the Spring components are not directly tied together anymore and can be tested in isolation.
```
Comment (DelfCo) [inline docs/spring-server/spring-bean-factory.md:10]: If you're going to introduce the idea that there are multiple pros and cons, you should really talk about them, and in terms that make it clear to a reader how they should weigh their own needs in light of either choice. (In other words, put it in terms of, "if you're doing X or Y, or Z is important to you, choose method A. If you're worried more about W, or you use U or T, then choose method B. If you like ice cream, then you must use method B.")

If you don't want to go into that, then just lose the "Both methods have their pros and cons, but" part, and leave the rest.

Comment (smyrick) [reply docs/spring-server/spring-bean-factory.md]: I took your advice and just consolidated the docs into a single page for multiple ways you can use nested arguments

File: docs/writing-schemas/nested-arguments.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
+---
+id: nested-arguments
+title: Nested Arguments
```
Comment (dariuszkuc) [inline docs/writing-schemas/nested-arguments.md:3]: i think this should be renamed to nested fields?

Comment (DelfCo) [reply docs/writing-schemas/nested-arguments.md:3]: For ease of maintenance going forward, it's easiest if the id and file name both match the title. Makes it easier to find the right file when users report a bug based on the page title, which is the only part they can see. You'll have to update links to this page if you rename it. You'll have to update website/sidebars.json if you change the id.

In fact, you need to add this to website/sidebars.json in any case, it's not there yet.

**Pull request merged**